pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -1,26 +1,21 @@
-{ _7zz
-, buildDotnetModule
-, copyDesktopItems
-, desktop-file-utils
-, dotnetCorePackages
-, fetchFromGitHub
-, fontconfig
-, lib
-, libICE
-, libSM
-, libX11
-, nexusmods-app
-, runCommand
-, enableUnfree ? false # Set to true to support RAR format mods
+{
+  lib,
+  _7zz,
+  buildDotnetModule,
+  copyDesktopItems,
+  desktop-file-utils,
+  dotnetCorePackages,
+  fetchFromGitHub,
+  fontconfig,
+  libICE,
+  libSM,
+  libX11,
+  nexusmods-app,
+  runCommand,
+  pname ? "nexusmods-app",
```

Comment:
Why would I want to optionally change the package name to an arbitrary string, when in reality there's only two useful package names? I'm pretty sure the original pattern of using `enableUnfree` (or something like it) is how it's done elsewhere. Having to override `_7zz` is much more technically difficult than setting `enableUnfree = true`. And finally, `enableUnfree` means we can easily add more optional unfree components if necessary without changing the function signature.

Reply:
Regarding `pname` as a package argument: that was discussed in a few comments above, mostly from here onwards: https://github.com/NixOS/nixpkgs/pull/331355#issuecomment-2262888714

Regarding the motivation for dropping `enableUnfree`, see this comment: https://github.com/NixOS/nixpkgs/pull/331355#issuecomment-2262848081

> Having to override _7zz is much more technically difficult than setting enableUnfree = true.

I don't agree that `_7zz = pkgs _7zz-rar` is **significantly** more difficult than `enableUnfree = true`.

I also prefer how the former is intrinsically clear about its purpose.

At first glance, the later could be interpreted as any number of unfree things being enabled.

> And finally, enableUnfree means we can easily add more optional unfree components if necessary without changing the function signature.

Maybe I'll eat my words, but I don't see that as a likely scenario. Nexusmods-app is committed to being an open source project with open source dependencies.

It is just unfortunate that one  optional feature of one dependency currently requires an unfree license.

If there were multiple unfree features to be enabled, then an `enableUnfree` argument would be more useful, I agree. If that happens we can revisit this 😀

Reply:
> I also prefer how the former is intrinsically clear about its purpose.

I prefer how the latter is more clear about the thing I care about. But of course people care about different things. :grin:

I still prefer the original design, but I'll let other reviewers decide.